samples: samples/subsys/usb/uvc : build failure on many NXP platforms

### Describe the bug

- frdm_mcxc242/mcxc242
- mimxrt685_evk/mimxrt685s/cm33
- mimxrt1180_evk/mimxrt1189/cm33
- mimxrt1170_evk@B/mimxrt1176/cm7
- mimxrt1170_evk@A/mimxrt1176/cm7
- mimxrt1060_evk@C/mimxrt1062/qspi
- frdm_mcxc444/mcxc444
- mimxrt1060_evk@A/mimxrt1062/qspi
- mimxrt1050_evk/mimxrt1052/hyperflash
- lpcxpresso55s36/lpc55s36
- frdm_k64f/mk64f12
- frdm_mcxa156/mcxa156
- frdm_mcxn236/mcxn236

0e3de8e3cac9a813dfa00cdfb0e0b4b3952f6125 is the first bad commit
commit 0e3de8e3cac9a813dfa00cdfb0e0b4b3952f6125
Author: Alain Volmat <alain.volmat@foss.st.com>
Date:   Wed Oct 22 11:58:36 2025 +0200

    usb: uvc: use struct uvc_frame_descriptor in most of function parameters
    
    This commit prepares introduction of the UVC Frame Based support by
    using the struct uvc_frame_descriptor as parameter of most of the UVC
    functions. struct uvc_frame_descriptor contains the common fields for
    all supported frame type and then depending on the DescriptorSubtype
    the pointer is casted in the correct struct definition.
    
    Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

 subsys/usb/device_next/class/usbd_uvc.c | 97 +++++++++++++++++++++------------
 subsys/usb/device_next/class/usbd_uvc.h |  5 +-
 2 files changed, 64 insertions(+), 38 deletions(-)


### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b frdm_k64f samples/subsys/usb/uvc
```

### Relevant log output

```shell
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: 
note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) 
						| ^~~~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:5211:9: 
note: in expansion of macro 'COND_CODE_1' 5211 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \
						| ^~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/display/display_st7796s.c:377:35: 
note: in expansion of macro 'MIPI_DBI_SPI_CONFIG_DT' 377
|.config = MIPI_DBI_SPI_CONFIG_DT( \ 
| ^~~~~~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/mcx_n9xx_evk_mcxn947_cpu0/zephyr/tests/drivers/display/display_check/tests.drivers.display.check.shield/zephyr/include/generated/zephyr/devicetree_generated.h:29227:51:
note: in expansion of macro 'ST7796S_INIT'29227 
| #define DT_FOREACH_OKAY_INST_sitronix_st7796s(fn) fn(0)
| ^~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36:
note: in expansion of macro 'DT_FOREACH_OKAY_INST_sitronix_st7796s' 105 
| #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/display/display_st7796s.c:414:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 414
| DT_INST_FOREACH_STATUS_OKAY(ST7796S_INIT) 
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.
```

### Impact

Major – Severely degrades functionality; workaround is difficult or unavailable.

### Environment

- OS: ubuntu
- Toolchain: Zephyr
- Version: v4.3.0-rc1

### Additional Context

_No response_

--- a/samples/subsys/usb/uvc/sample.yaml
+++ b/samples/subsys/usb/uvc/sample.yaml
@@ -34,8 +34,8 @@ tests:
     extra_configs:
       - CONFIG_VIDEO_ENCODER_H264=y
     extra_args:
-      - EXTRA_DTC_OVERLAY_FILE="app_h264enc.overlay"
-      - SHIELD=st_b_cams_imx_mb1854
+      - platform:stm32n6570_dk/stm32n657xx/sb:EXTRA_DTC_OVERLAY_FILE="app_h264enc.overlay"
+      - platform:stm32n6570_dk/stm32n657xx/sb:SHIELD=st_b_cams_imx_mb1854
     filter: dt_chosen_enabled("zephyr,camera")
     integration_platforms:
       - stm32n6570_dk/stm32n657xx/sb
@@ -46,8 +46,8 @@ tests:
     extra_configs:
       - CONFIG_VIDEO_ENCODER_JPEG=y
     extra_args:
-      - EXTRA_DTC_OVERLAY_FILE="app_jpegenc.overlay"
-      - SHIELD=st_b_cams_imx_mb1854
+      - platform:stm32n6570_dk/stm32n657xx/sb:EXTRA_DTC_OVERLAY_FILE="app_jpegenc.overlay"
+      - platform:stm32n6570_dk/stm32n657xx/sb:SHIELD=st_b_cams_imx_mb1854
     filter: dt_chosen_enabled("zephyr,camera")
     integration_platforms:
       - stm32n6570_dk/stm32n657xx/sb

